### ISSUE TITLE
ihaskell failed to install

### ISSUE THREAD

**Issue by Kwezan** [gibiansky/IHaskell#641]
Hi all,

I have tried to build ihaskell  for several times without success.
It seems that glib-0.13.2.1 can't be installed and whole process crush.
Did you tell me what going on???
I am running OS X 10.11.3, Python 3.5.1 and GHC 7.10.2.

Regards
Kwezan

MacBook-Pro-Zbigniew:IHaskell elzab$ ./build.sh all
[1 of 1] Compiling Main             ( dist/setup/setup.hs, dist/setup/Main.o ) [flags changed]
Linking ./dist/setup/setup ...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
cleaning...
CMD: cabal install --constraint arithmoi -llvm -j ./ghc-parser ./ipython-kernel ./. ./ihaskell-display/ihaskell-aeson ./ihaskell-display/ihaskell-basic ./ihaskell-display/ihaskell-blaze ./ihaskell-display/ihaskell-charts ./ihaskell-display/ihaskell-diagrams ./ihaskell-display/ihaskell-hatex ./ihaskell-display/ihaskell-juicypixels ./ihaskell-display/ihaskell-magic ./ihaskell-display/ihaskell-plot ./ihaskell-display/ihaskell-rlangqq ./ihaskell-display/ihaskell-static-canvas --force-reinstalls --max-backjumps=-1 --reorder-goals
Resolving dependencies...
Warning: Note that reinstalls are always dangerous. Continuing anyway...
Configuring ghc-parser-0.1.8.0...
Configuring glib-0.13.2.1...
Configuring ipython-kernel-0.8.3.0...
Building ipython-kernel-0.8.3.0...
Building glib-0.13.2.1...
Building ghc-parser-0.1.8.0...
Failed to install glib-0.13.2.1
Build log ( /Users/elzab/.cabal/logs/glib-0.13.2.1.log ):
[1 of 2] Compiling SetupWrapper     ( /var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/cabal-tmp-4473/glib-0.13.2.1/SetupWrapper.hs, /var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/cabal-tmp-4473/glib-0.13.2.1/dist/setup/SetupWrapper.o )
[2 of 2] Compiling Main             ( /var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/cabal-tmp-4473/glib-0.13.2.1/dist/setup/setup.hs, /var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/cabal-tmp-4473/glib-0.13.2.1/dist/setup/Main.o )
Linking /var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/cabal-tmp-4473/glib-0.13.2.1/dist/setup/setup ...
[1 of 2] Compiling Gtk2HsSetup      ( Gtk2HsSetup.hs, dist/setup-wrapper/Gtk2HsSetup.o )
[2 of 2] Compiling Main             ( SetupMain.hs, dist/setup-wrapper/Main.o )
Linking dist/setup-wrapper/setup ...
Configuring glib-0.13.2.1...
Building glib-0.13.2.1...
Preprocessing library glib-0.13.2.1...
[ 1 of 21] Compiling System.Glib.Attributes ( System/Glib/Attributes.hs, dist/build/System/Glib/Attributes.dyn_o )
[ 2 of 21] Compiling System.Glib.Flags ( System/Glib/Flags.hs, dist/build/System/Glib/Flags.dyn_o )
[ 3 of 21] Compiling System.Glib.FFI  ( System/Glib/FFI.hs, dist/build/System/Glib/FFI.dyn_o )
[ 4 of 21] Compiling System.Glib.Types ( dist/build/System/Glib/Types.hs, dist/build/System/Glib/Types.dyn_o )
[ 5 of 21] Compiling System.Glib.GString ( dist/build/System/Glib/GString.hs, dist/build/System/Glib/GString.dyn_o )
[ 6 of 21] Compiling System.Glib.GList ( dist/build/System/Glib/GList.hs, dist/build/System/Glib/GList.dyn_o )
[ 7 of 21] Compiling System.Glib.GType ( dist/build/System/Glib/GType.hs, dist/build/System/Glib/GType.dyn_o )
[ 8 of 21] Compiling System.Glib.GValue ( dist/build/System/Glib/GValue.hs, dist/build/System/Glib/GValue.dyn_o )
[ 9 of 21] Compiling System.Glib.GParameter ( dist/build/System/Glib/GParameter.hs, dist/build/System/Glib/GParameter.dyn_o )

System/Glib/GParameter.hsc:42:10: Warning:
    No explicit implementation for
      either ‘peek’ or ‘peekElemOff’ or ‘peekByteOff’
    In the instance declaration for ‘Storable GParameter’
[10 of 21] Compiling System.Glib.GTypeConstants ( dist/build/System/Glib/GTypeConstants.hs, dist/build/System/Glib/GTypeConstants.dyn_o )
[11 of 21] Compiling System.Glib.UTFString ( System/Glib/UTFString.hs, dist/build/System/Glib/UTFString.dyn_o )
[12 of 21] Compiling System.Glib.GObject ( dist/build/System/Glib/GObject.hs, dist/build/System/Glib/GObject.dyn_o )
[13 of 21] Compiling System.Glib.GValueTypes ( dist/build/System/Glib/GValueTypes.hs, dist/build/System/Glib/GValueTypes.dyn_o )
[14 of 21] Compiling System.Glib.StoreValue ( dist/build/System/Glib/StoreValue.hs, dist/build/System/Glib/StoreValue.dyn_o )
[15 of 21] Compiling System.Glib.Signals ( dist/build/System/Glib/Signals.hs, dist/build/System/Glib/Signals.dyn_o )
[16 of 21] Compiling System.Glib.MainLoop ( dist/build/System/Glib/MainLoop.hs, dist/build/System/Glib/MainLoop.dyn_o )
ld: -rpath can only be used when creating a dynamic final linked image
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Installed ghc-parser-0.1.8.0
Installed ipython-kernel-0.8.3.0
Configuring ihaskell-0.8.3.0...
Building ihaskell-0.8.3.0...
Installed ihaskell-0.8.3.0
Configuring ihaskell-aeson-0.3.0.0...
Configuring ihaskell-basic-0.3.0.0...
Configuring ihaskell-blaze-0.3.0.0...
Configuring ihaskell-charts-0.3.0.0...
Building ihaskell-basic-0.3.0.0...
Building ihaskell-blaze-0.3.0.0...
Building ihaskell-charts-0.3.0.0...
Building ihaskell-aeson-0.3.0.0...
Installed ihaskell-basic-0.3.0.0
Configuring ihaskell-hatex-0.2.1.0...
Configuring ihaskell-juicypixels-1.1.0.0...
Installed ihaskell-blaze-0.3.0.0
Configuring ihaskell-magic-0.3.0.0...
Installed ihaskell-charts-0.3.0.0
Configuring ihaskell-static-canvas-0.1.0.0...
Installed ihaskell-aeson-0.3.0.0
Building ihaskell-hatex-0.2.1.0...
Building ihaskell-juicypixels-1.1.0.0...
Installed ihaskell-hatex-0.2.1.0
Configuring ihaskell-rlangqq-0.3.0.0...
Building ihaskell-magic-0.3.0.0...
Building ihaskell-static-canvas-0.1.0.0...
Installed ihaskell-juicypixels-1.1.0.0
Installed ihaskell-magic-0.3.0.0
Building ihaskell-rlangqq-0.3.0.0...
Installed ihaskell-static-canvas-0.1.0.0
Installed ihaskell-rlangqq-0.3.0.0
cabal: Error: some packages failed to install:
diagrams-cairo-1.3.0.5 depends on glib-0.13.2.1 which failed to install.
glib-0.13.2.1 failed during the building phase. The exception was:
ExitFailure 1
ihaskell-diagrams-0.3.1.0 depends on glib-0.13.2.1 which failed to install.
ihaskell-plot-0.3.0.0 depends on glib-0.13.2.1 which failed to install.
pango-0.13.1.0 depends on glib-0.13.2.1 which failed to install.
plot-0.2.3.4 depends on glib-0.13.2.1 which failed to install.

**Comment by sumitsahrawat** [comment]
Did you try [stack](haskellstack.org)?

**Comment by Kwezan** [comment]
no - is it the difference?
I have tried already to install manually through cabal 
and running macos-install.sh and build.sh scripts.

**Comment by sumitsahrawat** [comment]
Well, it looks like an issue when building glib. Stack should solve that by providing a tested build.

**Comment by Kwezan** [comment]
no luck with stack ;)
There is another kind of errors.
All mentioned missing modules are installed.

MacBook-Pro-Zbigniew:IHaskell elzab$ stack install
blaze-markup-0.7.0.2: configure
blaze-markup-0.7.0.2: build
ansi-terminal-0.6.2.1: configure
ansi-terminal-0.6.2.1: build
asn1-types-0.3.0: configure
asn1-types-0.3.0: build
MonadRandom-0.4: configure
MonadRandom-0.4: build
ansi-terminal-0.6.2.1: copy/register
Progress: 4/117
--  While building package ansi-terminal-0.6.2.1 using:
      /Users/elzab/.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.22.4.0-ghc-7.10.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.4.0 register
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/Elzab/IHaskell/.stack-work/logs/ansi-terminal-0.6.2.1.log

    Configuring ansi-terminal-0.6.2.1...
    Building ansi-terminal-0.6.2.1...
    Preprocessing library ansi-terminal-0.6.2.1...
    [1 of 3] Compiling System.Console.ANSI.Common ( System/Console/ANSI/Common.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/System/Console/ANSI/Common.o )
    [2 of 3] Compiling System.Console.ANSI.Unix ( System/Console/ANSI/Unix.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/System/Console/ANSI/Unix.o )
    
    includes/Common-Include.hs:2:1: Warning:
        The import of ‘Control.Applicative’ is redundant
          except perhaps to import instances from ‘Control.Applicative’
        To import instances alone, use: import Control.Applicative()
    [3 of 3] Compiling System.Console.ANSI ( System/Console/ANSI.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/System/Console/ANSI.o )
    In-place registering ansi-terminal-0.6.2.1...
    Installing library in
    /Users/elzab/.stack/snapshots/x86_64-osx/nightly-2015-08-15/7.10.2/lib/x86_64-osx-ghc-7.10.2/ansi-terminal-0.6.2.1-0MG2i0b3psx7kYIvtaxDCd
    setup-Simple-Cabal-1.22.4.0-ghc-7.10.2: '/usr/local/bin/ghc' exited with an
    error:
    Bad interface file:
    .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/System/Console/ANSI.hi
    mismatched interface file ways (wanted "dyn", got "")


--  While building package MonadRandom-0.4 using:
      /Users/elzab/.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.22.4.0-ghc-7.10.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.4.0 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/Elzab/IHaskell/.stack-work/logs/MonadRandom-0.4.log

    Configuring MonadRandom-0.4...
    Building MonadRandom-0.4...
    Preprocessing library MonadRandom-0.4...
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack21863/MonadRandom-0.4/Control/Monad/Random.hs:52:18:
        Could not find module ‘Control.Monad.Cont’
        There are files missing in the ‘mtl-2.2.1@mtl_KMpng31YRYc5JfMWFZ3FCU’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack21863/MonadRandom-0.4/Control/Monad/Random.hs:53:18:
        Could not find module ‘Control.Monad.Error’
        There are files missing in the ‘mtl-2.2.1@mtl_KMpng31YRYc5JfMWFZ3FCU’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack21863/MonadRandom-0.4/Control/Monad/Random.hs:54:18:
        Could not find module ‘Control.Monad.Identity’
        There are files missing in the ‘mtl-2.2.1@mtl_KMpng31YRYc5JfMWFZ3FCU’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.
    
... and so on.

**Comment by sumitsahrawat** [comment]
That's unexpected. I think that you had some packages already installed, which have been damaged due to some reason. Can you give this a try?

1. Rename or remove `~/.cabal`, `~/.stack` and `/path/to/repo/.stack-work`
2. Run `stack setup` inside the repo
3. Run `setup install ihaskell`, it will install ihaskell without the display packages

**Comment by sumitsahrawat** [comment]
Also add `~/.ghc` to that list.

**Comment by Kwezan** [comment]
Done. No positive results.


MacBook-Pro-Zbigniew:IHaskell elzab$ stack setup
Downloaded nightly-2015-08-15 build plan.    
Caching build plan
Fetched package index.                                                                                    
Populated index cache.    
stack will use the GHC on your PATH
For more information on paths, see 'stack path' and 'stack exec env'
To use this GHC and packages outside of a project, consider using:
stack ghc, stack ghci, stack runghc, or stack exec

MacBook-Pro-Zbigniew:IHaskell elzab$ stack install ihaskell
[1 of 1] Compiling Main             ( /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/Setup.hs, /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/Setup.o )
Linking /Users/elzab/.stack/setup-exe-cache/x86_64-osx/tmp-setup-Simple-Cabal-1.22.4.0-ghc-7.10.2 ...
ansi-terminal-0.6.2.1: download
case-insensitive-1.2.0.4: download
asn1-types-0.3.0: download
cpphs-1.19.2: download
asn1-types-0.3.0: configure
asn1-types-0.3.0: build
ansi-terminal-0.6.2.1: configure
ansi-terminal-0.6.2.1: build
case-insensitive-1.2.0.4: configure
case-insensitive-1.2.0.4: build
cpphs-1.19.2: configure
cpphs-1.19.2: build
ansi-terminal-0.6.2.1: copy/register
Progress: 4/35
--  While building package ansi-terminal-0.6.2.1 using:
      /Users/elzab/.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.22.4.0-ghc-7.10.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.4.0 register
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/Elzab/IHaskell/.stack-work/logs/ansi-terminal-0.6.2.1.log

    Configuring ansi-terminal-0.6.2.1...
    Building ansi-terminal-0.6.2.1...
    Preprocessing library ansi-terminal-0.6.2.1...
    [1 of 3] Compiling System.Console.ANSI.Common ( System/Console/ANSI/Common.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/System/Console/ANSI/Common.o )
    [2 of 3] Compiling System.Console.ANSI.Unix ( System/Console/ANSI/Unix.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/System/Console/ANSI/Unix.o )
    
    includes/Common-Include.hs:2:1: Warning:
        The import of ‘Control.Applicative’ is redundant
          except perhaps to import instances from ‘Control.Applicative’
        To import instances alone, use: import Control.Applicative()
    [3 of 3] Compiling System.Console.ANSI ( System/Console/ANSI.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/System/Console/ANSI.o )
    In-place registering ansi-terminal-0.6.2.1...
    Installing library in
    /Users/elzab/.stack/snapshots/x86_64-osx/nightly-2015-08-15/7.10.2/lib/x86_64-osx-ghc-7.10.2/ansi-terminal-0.6.2.1-0MG2i0b3psx7kYIvtaxDCd
    setup-Simple-Cabal-1.22.4.0-ghc-7.10.2: '/usr/local/bin/ghc' exited with an
    error:
    Bad interface file:
    .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/System/Console/ANSI.hi
    mismatched interface file ways (wanted "dyn", got "")


--  While building package cpphs-1.19.2 using:
      /Users/elzab/.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.22.4.0-ghc-7.10.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.4.0 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/Elzab/IHaskell/.stack-work/logs/cpphs-1.19.2.log

    Configuring cpphs-1.19.2...
    Building cpphs-1.19.2...
    Preprocessing library cpphs-1.19.2...
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/cpphs-1.19.2/Language/Preprocessor/Cpphs/CppIfdef.hs:21:8:
        Could not find module ‘Text.Parse’
        There are files missing in the ‘polyparse-1.11@polyp_5gbM4m6k3rhGIQyEboR1bR’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/cpphs-1.19.2/Language/Preprocessor/Cpphs/MacroPass.hs:32:8:
        Could not find module ‘System.Time’
        There are files missing in the ‘old-time-1.1.0.3@oldti_FBWJJDS5FKr7UKFScBcUW2’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/cpphs-1.19.2/Language/Preprocessor/Cpphs/MacroPass.hs:33:8:
        Could not find module ‘System.Locale’
        There are files missing in the ‘old-locale-1.0.0.7@oldlo_D6X1KPq5Sui5XjrHMwvFwK’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.


--  While building package case-insensitive-1.2.0.4 using:
      /Users/elzab/.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.22.4.0-ghc-7.10.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.4.0 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/Elzab/IHaskell/.stack-work/logs/case-insensitive-1.2.0.4.log

    Configuring case-insensitive-1.2.0.4...
    Building case-insensitive-1.2.0.4...
    Preprocessing library case-insensitive-1.2.0.4...
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/case-insensitive-1.2.0.4/Data/CaseInsensitive/Internal.hs:59:18:
        Could not find module ‘Data.Text’
        There are files missing in the ‘text-1.2.1.3@text_1l1AN4I48k37RaQ6fm6CEh’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/case-insensitive-1.2.0.4/Data/CaseInsensitive/Internal.hs:60:18:
        Could not find module ‘Data.Text.Lazy’
        There are files missing in the ‘text-1.2.1.3@text_1l1AN4I48k37RaQ6fm6CEh’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/case-insensitive-1.2.0.4/Data/CaseInsensitive/Internal.hs:66:8:
        Could not find module ‘Data.Hashable’
        There are files missing in the ‘hashable-1.2.3.3@hasha_2GTZqecVp8NJZrNuKzGxfy’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.


--  While building package asn1-types-0.3.0 using:
      /Users/elzab/.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.22.4.0-ghc-7.10.2 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.4.0 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/Elzab/IHaskell/.stack-work/logs/asn1-types-0.3.0.log

    Configuring asn1-types-0.3.0...
    Building asn1-types-0.3.0...
    Preprocessing library asn1-types-0.3.0...
    
    /private/var/folders/9s/nctpw4516rjgy89wgkdtvwbc0000gn/T/stack24863/asn1-types-0.3.0/Data/ASN1/Types.hs:23:8:
        Could not find module ‘Data.Hourglass’
        There are files missing in the ‘hourglass-0.2.9@hourg_6Wf2z6eXMVzIFxt1igqsEA’ package,
        try running 'ghc-pkg check'.
        Use -v to see a list of the files searched for.
MacBook-Pro-Zbigniew:IHaskell elzab$

**Comment by Kwezan** [comment]
I will remove all previous ihaskell installations and files tomorrow and after 
that I make the new clean installation with stack.

**Comment by sumitsahrawat** [comment]
Well, there's one more thing you can try. Change the resolver in `stack.yaml` to `lts-5.3`. IHaskell is a part of that as well: https://www.stackage.org/lts-5.3/package/ihaskell-0.8.3.0

**Comment by sumitsahrawat** [comment]
It will install a new version of ghc for you though. You might want to try using `lts-3.22` first, if you want to use your system installed ghc.

**Comment by Kwezan** [comment]
OK I will do both. Thanks for your help.

**Comment by Kwezan** [comment]
Hi sumitsahrawat,

I have changed the resolver to lts-5.3, make a download of new ghc-7.10.3 and ..... voila it works!!!
Thank you for your suggestions!!!
Now I have  working executable in ~/.local/bin/ihaskell. 
Where I should move this exec in order to run ' jupyter notebook' rather instead of ' stack exec jupyter notebook' ?

**Comment by sumitsahrawat** [comment]
That should not be required. Once you do `stack exec ihaskell -- install --stack`, just a simple `jupyter notebook` should work, I think. If it causes an issue, you can follow the procedure below.

---

The best way would be to add the following line to your shell's RC file, `~/.bashrc` for bash, `~/.zshrc` for zsh and so on:

```bash
export PATH="$HOME/.local/bin:$PATH"
```

This will make all executables in `~/.local/bin` directly executable from the command line.

**Comment by Kwezan** [comment]
I have done it before with export PATH as above. This solution does not work perfectly however.
The problem is that if I run "jupyter notebook' from the command line and open the .ipnb notebook, the kernel is crashing down every time so I have to restart it manually.
When I am doing the same with 'stack exec jupyter notebook'  - everything is perfectly OK.

**Comment by Kwezan** [comment]
In the first case I get the message 'ihaskell: Not in scope: ‘IHaskell.Eval.Widgets.relayWidgetMessages’
in bash shell.
